lib/rules/eqeqeq.js

```diff
@@ -20,13 +20,30 @@ module.exports = {
         schema: [
             {
                 enum: ["always", "smart", "allow-null"]
+            },
+            {
+                type: "object",
+                properties: {
+                    null: {
+                        enum: ["always", "never", "ignore"]
+                    }
+                },
+                additionalProperties: false
             }
```

Comment:
`null` property can only be set if the rule is set to `always` correct? In that case, I think we should go with @mysticatea suggestion. It would not allow setting `null` property with either `smart` or `allow-null` options.